### DISCUSSION TITLE
How do I use a paramter in the url to pre-filter the results?

### DISCUSSION THREAD

**Discussion by mikezimm** [microsoft-search/pnp-modern-search#1762] [Q&A] [answered]
**Describe the solution you'd like**
I didn't see anything in the docs so I apologize if I missed something obvious.

I'd like the results (or via the search box) to be prefiltered by a paramter in the URL.

## So if I load my search page using the following Url....:
https://tenant.sharepoint.com/sites/Search/SitePages/Search.aspx?query=find me

I want the page act as if I typed "find me" in the search box... just load the page the first time with that search query.

**Describe alternatives you've considered**
I tested the ootb search function and when I hit enter, it adds:  ?q=find me to the Url in the search page.
When I added this to my test page url in the browser, it does not load the page with my query.
It just loads as if I didn't pass in any pre-filtered query.

**Comment by wobba** [comment]
See https://microsoft-search.github.io/pnp-modern-search/usage/search-box/#connections and https://microsoft-search.github.io/pnp-modern-search/usage/search-results/connections/

And take a look at the tutorials https://microsoft-search.github.io/pnp-modern-search/scenarios/

**Comment by mikezimm** [comment, marked as answer]
@wobba ,
Thanks for the info.  I missed some of the fine print.

Just for my documentation purposes and maybe someone else down the road,
What I think I missed was knowing that the webpart's "queryParameter's properties" are actually smart enough to automatically pick up the parameters on the current page and not a case of me typing in a the query parameter I wanted.

Once I added the query part to the url, the webpart gives me a dropdown to pick it.  Very nice!
And it works perfectly.

Thanks!

![image](https://user-images.githubusercontent.com/49648086/151564535-6aab2c11-6f55-4b31-bc45-49d97d062400.png)

**Reply by wobba** [reply]
@mikezimm a good point.. and it's not being smart enough - you actually have to have the params you want to use in the URL when you hit edit. If not you cannot pick them. So I'd say the SPFx dynamic data setup is a bit faulty in this regards - and we reuse what is provided by the framework.
